readme.txt
Kingmailer WordPress SMTP
=====================

Contributors: kingmailer, chandersbs, krishnamoniz
Tags: wordpress smtp, mailgun smtp, sendgrid smtp, gmail smtp, smtp, kingmailer
Requires at least: 3.3
Tested up to: 5.6
Stable tag: 0.3.08
License: GPLv2 or later

Improve your WordPress SMTP: send emails to users' inbox using Kingmailer WordPress SMTP plugin.

== Description ==

### Kingmailer WordPress SMTP: the #1 WordPress SMTP plugin

Since 2017 Kingmailer WordPress SMTP has powered thousands of websites worldwide in delivering emails to **users' inbox**.

Kingmailer's mission is **affordable SMTP for Everyone**. We are working day and night to give the highest quality SMTP email delivery service to everyone. 

Kingmailer [WordPress SMTP](https://kingmailer.co/) plugin contains the important settings fields to manage your SMTP. You will need to add/verify your domain first in your Kingmailer-account.

#### AFFORDABLE PRICING
While you can use our service for free, upgrades only cost **$5 p/month**. This will help you save a lot in monthly costs and at the same time help you deliver your emails to users' inbox. We also accept Bitcoin.

Kingmailer WordPress SMTP is the most trusted email delivery platform by developers, webmasters, small business and startup owners.

#### TAKING CARE OF YOUR WORDPRESS SMTP
Kingmailer is packed full of features, designed for you to see all your email traffic, inbound and outbound email traffic. By going into your Kingmailer account you get access to all sorts of information. Some features are:

* Real-time results
* Email statistics
* Webhooks & integrations
* Flexible SMTP APIs
* Optimal delivery
* Message archive & Logs

#### HIGH INBOX DELIVERY

We are a transactional only WordPress SMTP email service, this means high inbox delivery. 

If your emails land in the junk / spam folder, check the following:

* Is your domain or IP blacklisted?
* Does your domain have a poor sending reputation?
* Is there some configuration error on your web server?

#### TRUST THE EXPERTS

Kingmailer is powered by a team of expert developers, sysadmins and SMTP consultants. They work constantly to stay at the cutting edge of WordPress SMTP and to improve the plugin with every release.

Kingmailer WordPress SMTP is the **only Wordpress SMTP plugin made by world-renowned SMTP experts**.


== Installation ==
Starting with Kingmailer WordPress SMTP consists of just two steps: installing and setting up the plugin. Kingmailer WordPress SMTP is designed to work with the wp_mail functionality and will relay all emails.

### INSTALL KINGMAILER WORDPRESS SMTP FROM WITHIN WORDPRESS

1. Visit the plugins page within your dashboard and select ‘Add New’;
1. Search for ‘Kingmailer WordPress SMTP’;
1. Activate Kingmailer WordPress SMTP from your Plugins page;
1. Go to ‘after activation’ below.

### INSTALL KINGMAILER WORDPRESS SMTP MANUALLY

1. Upload the ‘kingmailer-smtp’ folder to the /wp-content/plugins/ directory;
1. Activate the Kingmailer WordPress SMTP plugin through the ‘Plugins’ menu in WordPress;
1. Go to ‘after activation’ below.

### AFTER ACTIVATION

1. You should see Settings button;
1. Go through the settings page and set up the plugin for your site;
1. You’re done!

== Frequently Asked Questions ==

= Can I use Kingmailer WordPress SMTP to send spam/bulk marketing newsletters? =

No.

= Can I use Kingmailer WordPress SMTP to send marketing newsletters? =

No.

= What types of emails are allowed with Kingmailer WordPress SMTP? =

Currently only transactional emails. 

= What are transactional emails? =

These are emails usually generated by your WordPress site, some examples are: 

1. Account alerts (subscription ending, credit card expiring, etc.)
1. Abandoned cart notifications
1. Event notifications
1. Order confirmations and purchase receipts
1. Responses to customer service inquiries
1. Password reminders or resets
1. Requests for feedback
1. Subscription welcome emails

= Which fields are in the Kingmailer WordPress SMTP plugin? =

The Kingmailer WordPress SMTP plugin is very simple and has the following fields:

1. Your domain
1. From name
1. From address
1. API-key

= Do I need a Kingmailer account to make this work? =

Yes.

= How or where do I get a Kingmailer account? =

1. Visit [Kingmailer.co](https://Kingmailer.co) click on sign up.
1. Click on the link in the activation email to activate your account
1. You now have a Kingmailer-account.

= How much does Kingmailer cost? =

Kingmailer, in case you will upgrade, only costs **$5 p/mo**, you can sign up and test for free. No credit card required.

= Does Kingmailer WordPress SMTP work with WooCommerce? =

Yes. Kingmailer WordPress SMTP works perfectly with WooCommerce.

= Which plugins work with Kingmailer WordPress SMTP? =

Woocommerce
WPForms
Elementor Forms
Contact Form 7
Gravity Forms
Fast Secure Contact Form
Visual Forms Builder
Contact Form Builder
PlanSo Forms
Quform
MyMail Newsletter by revaxarts
SendPress Newsletters
WP HTML Mail
Email Templates
WordPress Leads
...etc

All plugins that uses the WordPress API wp_mail to send emails.

= Which port does Kingmailer support? =

The Kingmailer WordPress SMTP plugin is designed for port 587. Other ports that we support are 25, 2525 and 465.

= Is the connection encrypted? =

Yes. All emails are encypted with TLS.

= Does Kingmailer also accept Bitcoin? =

Yes.

= Testing the configuration fails when using the HTTP API =

Your web server may not allow outbound HTTP connections. Set `Use HTTP API` to "No", and fill out the configuration options to SMTP and test again. A different possibility may be: your server IP is blocked by Spamhaus or other services that regulates spam. Please check if your IP is not blacklisted. 

= Can Kingmailer WordPress SMTP be configured globally for WordPress Multisite? =

Kingmailer WordPress SMTP plugin is not yet tested with WordPress Multisite.

= The plain text alternative of my mail looks different from my template =

Plain text alternatives are for mail clients that cannot process HTML. As such, html tags are stripped from the plain text alternative. This is done before applying the wp_mail filters.

= How do I get support? =

Feel free to send us an email wordpress@kingmailer.co or start a topic in the support forum.

== Screenshots ==

1. Configuration options for using the Kingmailer HTTP API


== Changelog ==

0.3.08 (2021-02-06)
* Attachment fix

= 0.3.07 (2020-12-12)::
* Remove SMTP functionality

= 0.3.0 (2020-11-28):
* Add SMTP functionality

= 0.2 (2020-11-22): =
* Updated text

= 0.1 (2020-10-17): =
* Initial Release


CHANGELOG.md
Changelog
=========

0.3.08 (2021-02-06)
* Attachment fix

0.3.07 (2020-12-12):
* Remove SMTP

0.3.06 (2020-12-12):
* Bug fix (slug change)

0.3.05 (2020-12-12):
* Bug fix (slug change)

0.3.04 (2020-12-05):
* Updated text domain

0.3.03 (2020-12-05):
* Bug fixes

0.3.02 (2020-11-29):
* Updated text domain

0.3.01 (2020-11-29):
* Add language file

0.3.0 (2020-11-28):
* Add SMTP functionality

0.2 (2020-11-20):
* Change to PHPMailer approach
* Text updated

0.1 (2020-10-17):
* Initial Release


includes/options-page.php
<?php

	/*
	 * kingmailer-wordpress-plugin - Sending mail from Wordpress using Kingmailer
	 * Copyright (C) 2020 Krishna Moniz
	 * Copyright (C) 2016 Mailgun, et al.
	 *
	 * This program is free software; you can redistribute it and/or modify
	 * it under the terms of the GNU General Public License as published by
	 * the Free Software Foundation; either version 2 of the License, or
	 * (at your option) any later version.
	 *
	 * This program is distributed in the hope that it will be useful,
	 * but WITHOUT ANY WARRANTY; without even the implied warranty of
	 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	 * GNU General Public License for more details.
	 *
	 * You should have received a copy of the GNU General Public License along
	 * with this program; if not, write to the Free Software Foundation, Inc.,
	 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
	 */

?>
<div class="wrap">
	<div id="icon-options-general" class="icon32"><br/></div>

	<h2><?php _e('Kingmailer', 'kingmailer-smtp'); ?></h2>

	<p>
		<?php
			$url = 'https://kingmailer.co/';
			$link = sprintf(
				wp_kses(
					__('A <a href="%1$s" target="%2$s">Kingmailer</a> account is required to use this plugin and the Kingmailer service.', 'kingmailer-smtp'),
					array('a' => array(
							'href' => array(),
							'target' => array()
						)
					)
				), esc_url($url), '_blank'
			);
			echo $link;
		?>
	</p>

	<h3><?php _e('Configuration', 'kingmailer-smtp'); ?></h3>
	<form id="kingmailer-form" action="options.php" method="post">
		<?php settings_fields('kingmailer-smtp'); ?>

		<table class="form-table">
		<tr valign="top">
				<th scope="row">
					<?php _e('Your Domain', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[domain]"
						   value="<?php esc_attr_e($this->get_option('domain')); ?>"
						   placeholder="example.com"
					/>
					<p class="description">
						<?php
							$link = sprintf(
								wp_kses(
									__('The domain you have added in your Kingmailer-account. Visit <a href="%1$s" target="%2$s">Kingmailer.co</a> to sign up for an account.', 'kingmailer-smtp'),
									array('a' => array(
											'href' => array(),
											'target' => array()
										)
									)
								), esc_url($url), '_blank'
							);
							echo $link;
						?>
					</p>
				</td>
			</tr>
			<tr valign="top">
				<th scope="row">
					<?php _e('From Name', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[from-name]"
						   value="<?php esc_attr_e($this->get_option('from-name')); ?>"
						   placeholder="Excited User"
					/>
					<p class="description">
						<?php
							_e('The "User Name" part of the sender information. This can be your domain name.', 'kingmailer-smtp');
						?>
					</p>
				</td>
			</tr>
			<tr valign="top">
				<th scope="row">
					<?php _e('From Address', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text"
						   class="regular-text"
						   name="kingmailer-smtp[from-address]"
						   value="<?php esc_attr_e($this->get_option('from-address')); ?>"
						   placeholder="info@example.com"
					/>
					<p class="description">
						<?php
							_e('The &lt;info@example.com&gt; part of the sender information. This address will appear as the `From` address on sent mail. <strong>It is recommended that the @mydomain portion matches your Kingmailer sending domain.</strong>', 'kingmailer-smtp');
						?>
					</p>
				</td>
			</tr>
			<tr valign="top">
				<th scope="row">
					<?php _e('Override "From" Details', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<select name="kingmailer-smtp[override-from]">
						<option value="1"<?php selected('1', $this->get_option('override-from', null, '1')); ?>><?php _e('Yes', 'kingmailer-smtp'); ?></option>
						<option value="0"<?php selected('0', $this->get_option('override-from', null, '0')); ?>><?php _e('No', 'kingmailer-smtp'); ?></option>
					</select>
					<p class="description">
						<?php
							_e('If enabled, all emails will be sent with the above <code>From Name</code> and <code>From Address</code>, regardless of values set by other plugins. Useful for cases where other plugins don\'t play nice with our settings. Defaults to "Yes".', 'kingmailer-smtp');
						?>
					</p>
				</td>
			</tr>
			<!-- IMPORTANT remove this hidden field when reactivating SMTP -->
			<input type="hidden" id="kingmailer-api" name="kingmailer-smtp[use_api]" value="1">
			<!-- <tr valign="top">
				<th scope="row">
					<?php _e('Use HTTP API', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<select id="kingmailer-api" name="kingmailer-smtp[use_api]">
						<option value="1"<?php selected('1', $this->get_option('use_api')); ?>><?php _e('Yes', 'kingmailer-smtp'); ?></option>
						<option value="0"<?php selected('0', $this->get_option('use_api')); ?>><?php _e('No', 'kingmailer-smtp'); ?></option>
					</select>
					<p class="description">
						<?php
							_e('Set this to "No" if your server cannot make outbound HTTP connections or if emails are not being delivered. "No" will cause this plugin to use SMTP. Defaults to "Yes".', 'kingmailer-smtp');
						?>
					</p>
				</td>
			</tr> -->
			<tr valign="top" class="kingmailer-api">
				<th scope="row">
					<?php _e('API-host', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[api_host]"
						   value="<?php esc_attr_e($this->get_option('api_host')); ?>"
						   placeholder="kingmailer.org"
						   readonly
					/>
					<p class="description">
						<?php _e('Your Kingmailer API-host. Fixed at "kingmailer.org"', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr>
			<tr valign="top" class="kingmailer-api">
				<th scope="row">
					<?php _e('API Key', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text" name="kingmailer-smtp[api_key]"
						   value="<?php esc_attr_e($this->get_option('api_key')); ?>"
						   placeholder="ur7keAGRjYB3W5NfKQEm563Z"
					/>
					<p class="description">
						<?php _e('Your Kingmailer API key. Create an API by visiting the credentials section in your Kingmailer-account.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr>
			<tr valign="top" class="kingmailer-smtp">
				<th scope="row">
					<?php _e('SMTP-host', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[host]"
						   value="<?php esc_attr_e($this->get_option('host')); ?>"
						   placeholder="kingmailer.org"
						   readonly
					/>
					<p class="description">
						<?php _e('Your Kingmailer SMTP-host. Fixed at "kingmailer.org"', 'kingmailer-smtp'); ?>
						<br />
						<?php _e('Only valid for use with SMTP.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr>
			<!-- <tr valign="top" class="kingmailer-smtp">
				<th scope="row">
					<?php _e('Username', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[username]"
						   value="<?php esc_attr_e($this->get_option('username')); ?>"
						   placeholder="postmaster"
					/>
					<p class="description">
						<?php _e('Your Kingmailer SMTP username.', 'kingmailer-smtp'); ?>
						<br />
						<?php _e('Only valid for use with SMTP.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr>
			<tr valign="top" class="kingmailer-smtp">
				<th scope="row">
					<?php _e('Password', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[password]"
						   value="<?php esc_attr_e($this->get_option('password')); ?>"
						   placeholder="my-password"
					/>
					<p class="description">
						<?php _e('Your Kingmailer SMTP password that goes with the above username.', 'kingmailer-smtp'); ?>
						<br />
						<?php _e('Only valid for use with SMTP.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr>
			<tr valign="top" class="kingmailer-smtp">
				<th scope="row">
					<?php _e('Port number', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<input type="text" class="regular-text"
						   name="kingmailer-smtp[smtp_port]"
						   value="<?php esc_attr_e($this->get_option('smtp_port')); ?>"
						   placeholder="587"
					/>
					<p class="description">
						<?php _e('Specify which port to use for SMTP mailing. Kingmailer only allows for standard SMTP ports (25, 465, and 587). Defaults to "587"', 'kingmailer-smtp'); ?>
						<br />
						<?php _e('Only valid for use with SMTP.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr> -->
			<!-- <tr valign="top" class="kingmailer-smtp">
				<th scope="row">
					<?php // _e('Use Secure SMTP', 'kingmailer-smtp'); ?>
				</th>
				<td>
					<select name="kingmailer-smtp[secure]">
						<option value="1"<?php // selected('1', $this->get_option('secure')); ?>><?php // _e('Yes', 'kingmailer-smtp'); ?></option>
						<option value="0"<?php // selected('0', $this->get_option('secure')); ?>><?php // _e('No', 'kingmailer-smtp'); ?></option>
					</select>
					<p class="description">
						<?php // _e('Set this to "Yes" to send your mails over a secure SSL/TLS connection. If you set this to "No" your password will be sent in plain text. Default is "Yes".', 'kingmailer-smtp'); ?>
						<br />
						<?php // _e('Only valid for use with SMTP.', 'kingmailer-smtp'); ?>
					</p>
				</td>
			</tr> -->
		</table>
		<p>
			<?php
				_e('Please click <code>Save Changes</code> before attempting to test the configuration.', 'kingmailer-smtp');
			?>
		</p>
		<p class="submit">
			<input type="submit"
				   class="button-primary"
				   value="<?php _e('Save Changes', 'kingmailer-smtp'); ?>"
			/>
			<input type="button"
				   id="kingmailer-test"
				   class="button-secondary"
				   value="<?php _e('Send Test Mail', 'kingmailer-smtp'); ?>"
			/>
		</p>
	</form>
</div>
